src/libexpr/primops/fetchClosure.cc

```diff
@@ -92,82 +187,12 @@ static void prim_fetchClosure(EvalState & state, const PosIdx pos, Value * * arg
-            if (inputAddressed)
-                throw Error({
-                    .msg = hintfmt("The 'fetchClosure' result, '%s' is not input addressed, despite 'inputAddressed' being set to true. It is ok or even preferable to return a content addressed path, so you probably want to remove the 'inputAddressed' attribute",
-                        state.store->printStorePath(*toPath)),
-                    .errPos = state.positions[pos]
-                });
-            else
-                throw Error({
-                    .msg = hintfmt("The 'fetchClosure' result, '%s' is input addressed, despite 'inputAddressed' being set to false. Please make sure you passed the correct path(s) or set 'inputAddressed' to true if you intended to return an input addressed path",
-                        state.store->printStorePath(*toPath)),
-                    .errPos = state.positions[pos]
-                });
-        }
-    } else {
-        /*
-          While it's fine to omit the inputAddressed attribute for CA paths,
-          we want input addressing to be explicit, to inform readers and to give
-          expression authors an opportunity to improve their user experience;
-          see message below.
-        */
-        if (!info->isContentAddressed(*state.store)) {
-            if (enableRewriting) {
-                // We don't perform the rewriting when outPath already exists, as an optimisation.
-                // However, we can quickly detect a mistake if the toPath is input addressed.
-                // Ideally we'd compute the path for them here, but this error message is unlikely to occur in practice, so we keep it simple.
-                throw Error({
-                    .msg = hintfmt("The 'toPath' value '%s' is input addressed, so it can't possibly be the result of rewriting. You may set 'toPath' to an empty string to figure out the correct path.",
-                        state.store->printStorePath(*toPath)),
-                    .errPos = state.positions[pos]
-                });
-            } else {
-                // We just checked toPath, but we report fromPath, because that's what the user certainly passed.
-                assert (toPath == fromPath);
-                throw Error({
-                    .msg = hintfmt("The 'fromPath' value '%s' is input addressed, but input addressing was not requested. If you do intend to return an input addressed store path, add 'inputAddressed = true;' to the 'fetchClosure' arguments. Note that content addressing does not require users to configure a trusted binary cache public key on their systems, and is therefore preferred.",
-                        state.store->printStorePath(*fromPath)),
-                    .errPos = state.positions[pos]
-                });
-            }
-        }
-    }
-
-    state.mkStorePathString(*toPath, v);
+    if (enableRewriting)
+        runFetchClosureWithRewrite(state, pos, *fromStore, *fromPath, *toPath, v);
```

Comment:
Should be `toPath` not `*toPath` right?

Reply:
Ah well fixed in the next commit.

Reply:
Applied.
I might have made a mistake with autofixup.